### PR TITLE
About demand-driven architecture

### PR DESCRIPTION
**Pull request by ganwell**
I wrote this because the systems that inspired me to write chirp are demand-driven and demand-driven is just awesome, but....

### CODE REVIEW

**Comment by ganwell** [comment]
@winged ping

File: README.rst [hunk 1]
```diff
@@ -224,7 +224,7 @@ for cpp, so \*.h would not get checked.
 .. code-block:: bash
```
Comment (winged) [inline README.rst]: This does not look like a bash code block :) Does Sphinx support vim syntax?

File: README.rst [hunk 2]
```diff
@@ -243,6 +243,23 @@ If you use clang complete, we recommend
    let g:clang_snippets        = 1
    let g:clang_snippets_engine = 'clang_complete'
 
+FAQ
+===
+
+Why isn't chirp demand-driven?
+------------------------------
```
Comment (winged) [inline README.rst]: What is demand-driven? Maybe add a short intro or external link?

**Comment by ganwell** [comment]
@winged do you know a good link? When I search a get a bunch of research-papers nobody is going to read.

**Comment by winged** [comment]
@ganwell Even I don't know exactly what it is.. so maybe it's better to write a small paragraph and only tangentially link to the research papers.

**Comment by ganwell** [comment]
Note to me: Use the demand-driven ZeroMQ pattern and GenStage as examples.

**Pull request merged**